### ISSUE TITLE
Null SecurityStamp results in in always invalid token.

### ISSUE THREAD

**Issue by LindaLawton** [aspnet/Identity#2022]
If a user has a SecurityStamp of null and SupportsUserSecurityStamp is true.  The token is created with "" as the stamp.     [L82](https://github.com/aspnet/Identity/blob/c7276ce2f76312ddd7fccad6e399da96b9f6fae1/src/Identity/DataProtectionTokenProvider.cs#L82)

However when it is validated Its compared against null resulting in always false validation  [L135](https://github.com/aspnet/Identity/blob/c7276ce2f76312ddd7fccad6e399da96b9f6fae1/src/Identity/DataProtectionTokenProvider.cs#L135)

**Sugestions**

Either the create token should create a stamp for the user if SupportsUserSecurityStamp is true and the stamp is null.    Or the compare in the validation should check for null and set it to space if it is.

**Comment by blowdart** [comment]
If a stamp is support by the store it should never be null. So the right approach here would be to throw when a stamp is generated. Thoughts @HaoK?

**Comment by HaoK** [comment]
Yeah your users are in a broken state here basically.  We could be more defensive and throw when stamp is null in more places I guess, something to look at for 3.0?

**Comment by HaoK** [comment]
I don't think we should treat null security stamps as valid ever, my suggestion was rather that we have any methods throw saying user has a null security stamp instead of just always returning false/invalid.  

Basically users being broken should result in exceptions, @blowdart thoughts?

**Comment by LindaLawton** [comment]
Well the thing is if you check the validate method it throws all exceptions and errors away.   So throwing an exception and invalidating the token isnt helping at all.   That's my problem.  There is no logging as to what is wrong with the tokens you are leaving developers to guess.  This has lead me to create my own DataProtectorTokenProvider that includes LOGGING.   

I know you didn't ask for my thoughts but 

If you feel that all security stamps null users should not be able to validate a token then i would argue that they should not be able to create one either it should fail this would at least give the developer an idea that somethings wrong.

I also think you should add logging to give developers a clue whats wrong rather then just tossing everything.

**Comment by HaoK** [comment]
> If you feel that all security stamps null users should not be able to validate a token then i would argue that they should not be able to create one either.

Yup that's the idea, we should just throw anywhere we see a null security stamp when its enabled

**Comment by blowdart** [comment]
Indeed, throw everywhere :)

**Comment by blowdart** [comment]
Assigning to @HaoK for 3.0 so it can be consistent everywhere.
